CONTRIBUTING.md

```diff
@@ -1,3 +1,22 @@
 # Contributing
-If you want to contribute, feel free to fork and make a PR. Please lint and run `dartfmt` before opening a PR.
-And make sure to always make your PRs to `development` branch.
\ No newline at end of file
+Nyxx is free and open-source project, and all contributions are welcome and highly appreciated.
+However, please conform to the following guidelines when possible.
+
+## Development cycle
+
+All changes should be discussed beforehand either in issue or pull request on github 
+or in a discussion in our Discord channel with library regulars or other contributors.
+
+All issues marked with 'help-needed' badge are free to be picked up by any member of the community. 
+
+### Pull Requests
+
+Pull requests should be descriptive about changes that are made. 
+If adding new functionality or modifying existing, documentation should be added/modified to reflect changes.
+
+## Coding style
+
+We attempt to conform [Effective Dart Coding Style](https://dart.dev/guides/language/effective-dart/style) where possible.
```

Comment:
Hi, you can check out my package for enabling those lints .<3 https://github.com/tenhobi/effective_dart

Reply:
We do have lints enabled in: https://github.com/l7ssha/nyxx/blob/rewrite_modular/nyxx/analysis_options.yaml

Reply:
Yes, that's cool. Just wanted to offer a way how to use synced rule list. :-)